js/data/buffer/circle_vertex_buffer.js

```diff
@@ -1,41 +1,31 @@
 'use strict';
 
 var util = require('../../util/util');
-var Buffer = require('./buffer');
-
-module.exports = CircleVertexBuffer;
-
-/**
- * This contains the data that displays circle markers on the map,
- * including their centerpoint
- */
-function CircleVertexBuffer(buffer) {
-    Buffer.call(this, buffer);
+var Buffer2 = require('../buffer2');
+
+function CircleVertexBuffer(options) {
+    Buffer2.call(this, options || {
+        type: Buffer2.BufferType.VERTEX,
+        attributes: {
+            verticies: {
+                components: 4,
+                type: Buffer2.AttributeType.UNSIGNED_SHORT
+            }
+        }
+    });
 }
 
-CircleVertexBuffer.prototype = util.inherit(Buffer, {
-    defaultLength: 2048 * 16,
-
-    itemSize: 4, // 2 bytes per short * 4 of them
-
+CircleVertexBuffer.prototype = util.inherit(Buffer2, {
     add: function(x, y, extrudeX, extrudeY) {
-        var pos = this.pos,
-            pos2 = pos / 2;
-
-        this.resize();
-
-        // pack the extrusion of -1 or 1 into one short
-        this.shorts[pos2 + 0] = (x * 2) + ((extrudeX + 1) / 2);
-        this.shorts[pos2 + 1] = (y * 2) + ((extrudeY + 1) / 2);
-
-        this.pos += this.itemSize;
+        this.push([x, y, extrudeX, extrudeY]);
```

Comment:
Oh -- looks like you'll want to change this to 

```
this.push([
  (x * 2) + ((extrudeX + 1) / 2),
  (y * 2) + ((extrudeY + 1) / 2)
]);
```

... that probably explains the segfault.